app/models/openstax/accounts/account.rb

```diff
@@ -24,9 +24,12 @@ class Account < ActiveRecord::Base
                           :confirmed_faculty, :rejected_faculty]
 
     after_initialize :set_default_faculty_status
-
     validates :faculty_status, presence: true
 
+    enum role: [:unknown_role, :student, :instructor, :administrator, :librarian, :designer, :other]
+    after_initialize :set_default_role
+    validates :role, presence: true
```

Comment:
I don't think you need the `after_initialize` and `set_default_role`.  Since you're setting the default to 0 in the migration below, ActiveRecord will set that as the default for new records automatically.